lib/net.js

```diff
@@ -894,6 +906,12 @@ Socket.prototype.connect = function(options, cb) {
   self._connecting = true;
   self.writable = true;
 
+  // Cork then uncork upon connecting.
+  this.cork();
+  this.once('connect', function() {
+    this.uncork();
+  });
```

Comment:
By corking eagerly, we make sure (as often as possible) that the writes stay in the stream infrastructure rather than like before, where one sat in net.js.